version: "3.7"

x-app-defaults: &app-defaults
  restart: unless-stopped
  image: listmonk/listmonk:latest
  # ports:
  #   - "2000"
  networks:
    # - listmonk
    - traefik
  environment:
    - TZ=Etc/EST

x-db-defaults: &db-defaults
  image: postgres:13-alpine
  ports:
    - "5432"
  networks:
    - traefik
  environment:
    - POSTGRES_PASSWORD=samyogdb
    - POSTGRES_USER=listmonk
    - POSTGRES_DB=listmonk
  restart: unless-stopped
  healthcheck:
    test: [ "CMD-SHELL", "pg_isready -U listmonk" ]
    interval: 10s
    timeout: 5s
    retries: 6

services:
  db:
    <<: *db-defaults
    container_name: listmonk_db
    volumes:
      - type: volume
        source: listmonk-data
        target: /var/lib/postgresql/data
      - ./db_backups:/db_backups

  app:
    <<: *app-defaults
    container_name: listmonk_app
    depends_on:
      - db
    command: "./listmonk --static-dir=/listmonk/static"
    volumes:
      - ./config.toml:/listmonk/config.toml
      - ./uploads:/listmonk/uploads
      - ./static:/listmonk/static
    labels:
      - "traefik.http.routers.listmonk_app.rule=Host(`mailing.specux.com`)"
      - "traefik.http.services.listmonk_app.loadbalancer.server.port=2000"

networks:
  traefik:
    external: true
  # listmonk:


volumes:
  listmonk-data:
